### ISSUE TITLE
dnf5daemon: Add functions to enable/disable repos in org.rpm.dnf.v0.rpm.Repo

### ISSUE THREAD

**Issue by mcrha** [rpm-software-management/dnf5#771]
In order to be feature complete with PackageKit (on the gnome-software side), it would be good to add functions to enable and disable repositories into the org.rpm.dnf.v0.rpm.Repo D-Bus interface.

**Comment by jan-kolarik** [comment]
More details in https://bugzilla.redhat.com/show_bug.cgi?id=2124496.

**Comment by mcrha** [comment]
Maybe I do something wrong. I've compiled dnf5 at commit 440469681c8ec06884dcf8dbd73c2ab9105eeed1 and I run it (started as a root from a command line `$PREFIX/sbin/dnf5daemon-server --verbose` (the `--verbose` doesn't seem to do anything)), and I see it being used based on the `DEBUG` lines on the dnf5daemon-server terminal.

The response from enable/disable repo call is an error: `Action org.rpm.dnf.v0.rpm.Repo.conf_write is not registered`.

When I try to install an RPM app I'm asked for a wheel user password and the app is installed, from which i suppose my way of running the server is not that wrong. I've not been asked for the wheel user password when doing enable/disable of the repo.

I know I'm running the built version of the dnf5 server, because when the system server is running the error is different: `Unknown method disable or interface org.rpm.dnf.v0.rpm.Repo`, which makes sense, the method had been just added.

**Comment by ppisar** [comment]
Does your $PREFIX/sbin/dnf5daemon-server link to your new libdnf5.so.1?

I tried enabling a repository with "dnf5daemon-client repo enable f41-build" command with DNF5 built from commit c60cd47e5920fe22d06b0e70e96c6d3fa354be72 (later than yours) and it worked for me fine. You can try the nightly builds yourself from <https://download.copr.fedorainfracloud.org/results/rpmsoftwaremanagement/dnf-nightly/fedora-rawhide-x86_64/>.

**Comment by m-blaha** [comment]
I think the problem is that the PR (specifically 57f8f3a61ffc8de47602145bf10dae2cfb57c36d) renames polkit policy org.rpm.dnf.v0.rpm.RepoConf.write -> org.rpm.dnf.v0.rpm.Repo.conf_write.

**Comment by mcrha** [comment]
> Does your $PREFIX/sbin/dnf5daemon-server link to your new libdnf5.so.1?

Yes, it does. Though I see it depends how the terminal is set up, because `ldd` can pick the system lib as well.

> I think the problem is that the PR ... renames polkit policy

Aha. When I replace the system-installed polkit file with that in my $PREFIX, I am asked for the admin credentials when doing the repo enable, thus it's it.

It still fails [1], but it's another problem on my side, in my custom build. I consider this working. I'm sorry for the noise.

[1] I see this on the dnf5daemon-server console:

```
2024-03-11T17:47:15+0100 [26487] DEBUG Error reading repository configuration directory "/etc/distro.repos.d":
filesystem error: cannot make canonical path: No such file or directory [/etc/distro.repos.d]
2024-03-11T17:47:15+0100 [26487] DEBUG Error reading repository configuration directory "/usr/share/dnf5/repos.d":
filesystem error: cannot make canonical path: No such file or directory [/usr/share/dnf5/repos.d]
```

Interesting it is able to read from `/etc/yum.repos.d/` with no problem.
